source/fe/fe_nedelec_sz.cc

```diff
@@ -264,1302 +310,1151 @@ FE_NedelecSZ<dim, spacedim>::get_data(
-                    (flags & update_hessians) ?
-                      4 :
-                      ((flags & update_gradients) ? 3 : 2);
-
-                  std::vector<std::vector<double>> polyx(
-                    degree, std::vector<double>(poly_length));
-                  std::vector<std::vector<double>> polyy(
-                    degree, std::vector<double>(poly_length));
-                  for (unsigned int i = 0; i < degree; ++i)
-                    {
-                      // Compute all required 1d polynomials and their
-                      // derivatives, starting at degree 2. e.g. to access
-                      // L'_{3}(2x-1) use polyx[1][1].
-                      IntegratedLegendrePolynomials[i + 2].value(
-                        cell_points[q][0], polyx[i]);
-                      IntegratedLegendrePolynomials[i + 2].value(
-                        cell_points[q][1], polyy[i]);
-                    }
-                  // Now use these to compute the shape functions:
-                  if (flags & update_values)
-                    {
-                      for (unsigned int j = 0; j < degree; ++j)
-                        {
-                          const unsigned int shift_j(j * degree);
-                          for (unsigned int i = 0; i < degree; ++i)
-                            {
-                              const unsigned int shift_ij(i + shift_j);
 
-                              // Type 1:
-                              const unsigned int dof_index1(cell_type1_offset +
-                                                            shift_ij);
-                              data.shape_values[dof_index1][q][0] =
-                                2.0 * polyx[i][1] * polyy[j][0];
-                              data.shape_values[dof_index1][q][1] =
-                                2.0 * polyx[i][0] * polyy[j][1];
 
-                              // Type 2:
-                              const unsigned int dof_index2(cell_type2_offset +
-                                                            shift_ij);
-                              data.shape_values[dof_index2][q][0] =
-                                data.shape_values[dof_index1][q][0];
-                              data.shape_values[dof_index2][q][1] =
-                                -1.0 * data.shape_values[dof_index1][q][1];
-                            }
-                          // Type 3:
-                          const unsigned int dof_index3_1(cell_type3_offset1 +
-                                                          j);
-                          data.shape_values[dof_index3_1][q][0] = polyy[j][0];
-                          data.shape_values[dof_index3_1][q][1] = 0.0;
+template <>
+void
+FE_NedelecSZ<3, 3>::evaluate(
+  const std::vector<Point<3>> p_list,
+  const UpdateFlags           update_flags,
+  std::unique_ptr<typename dealii::FiniteElement<3, 3>::InternalDataBase>
+    &data_ptr) const
+{
```

Comment:
Oh, I see, you here split things off for the 3d case. Why is this necessary? I think you lose the `<2,3>` case this way.

Reply:
I just made this to get rid of `case(dim):`.

And in the file [`include/deal.II/fe/fe_nedelec_sz.h`](https://github.com/dealii/dealii/blob/31a1263477fc0d5b85bb7a2456ebb32744616593/include/deal.II/fe/fe_nedelec_sz.h#L86) there is the static assert: 

 ```
   static_assert(dim == spacedim,
                "FE_NedelecSZ is only implemented for dim==spacedim!");
```

So I haven´t toght about the `<2,3>`  case.
But with `<2,3>` in mind (eventhough it is not supported at the moment) it may be better to merge those two functions into one again.

What do you think?

Reply:
I've added a commit that merges these evaluate functions into one.
I keep that as an extra commit for the moment so that it can be undone. But if we want to keep it that way, I can rebase that commit into the initial commit.

Reply:
No, don't worry -- this is ok.